kubeflow/katib/vizier.libsonnet

```diff
@@ -16,10 +18,53 @@
     $.parts(params, namespace).uiClusterRole,
     $.parts(params, namespace).uiClusterRoleBinding,
     $.parts(params, namespace).uiServiceAccount,
-  ],
+  ] + if util.toBool(params.injectIstio) then [
+    $.parts(params, namespace).virtualService,
+  ] else [],
 
   parts(params, namespace):: {
 
+    virtualService: {
+      apiVersion: "networking.istio.io/v1alpha3",
+      kind: "VirtualService",
+      metadata: {
+        name: "katib-ui",
+        namespace: namespace,
+      },
+      spec: {
+        hosts: [
+          "*",
+        ],
+        gateways: [
+          "kubeflow-gateway",
+        ],
+        http: [
+          {
+            match: [
+              {
+                uri: {
+                  prefix: "/katib/",
+                },
+              },
+            ],
+            rewrite: {
+              uri: "/katib/",
+            },
+            route: [
+              {
+                destination: {
+                  host: "katib-ui." + namespace + ".svc.cluster.local",
```

Comment:
IMO in the most cases we should keep `cluster-domain` as it is?  could you share some information as why it might not be the case?

I do agree `host: "katib-ui"` will do but API actually recommends using full name: [link](https://github.com/istio/api/blob/6b8d1849e7f44ef523b4442af69b57ddc960d38b/networking/v1alpha3/virtual_service.proto#L330)

Reply:
The suffix "cluster.local" is a kubernetes setting so it won't always be "cluster.local"